commands/commands.go

```diff
@@ -339,7 +339,31 @@ func quote(ctx context.Context, s *discordgo.Session, m *discordgo.MessageCreate
 			log.WithFields(ctx.Value(logKey).(log.Fields)).Error("Error getting messages")
 			return
 		}
-		message := messages.Get(rand.Intn(messages.Len()))
+
+		// quote takes into consideration nubmer of reacts for considering which message to quote, more reactions means higher chance
+		// messageDefaultWeight is relative to weight of a single reaction
+		weightsSlice := make([]uint, messages.Len())
+		messageDefaultWeight := viper.Get("bot.quote.default_message_weight").(uint)
```

Comment:
This will throw a runtime error. This is because when the primitive int is initialized in the viper default, it's seen as an `int`. The type assertion here asserts it as a `unit` which will fail. See https://play.golang.org/p/esDGBPwu2Oh

Reply:
my bad it ran on my machine so id thought it be fine